src/typeEvaluator/typeEvaluate.ts

```diff
@@ -119,18 +119,40 @@ function handleDerefNode(node: DerefNode, scope: Scope): TypeNode {
 function mapObjectSplat(
   node: TypeNode,
   scope: Scope,
-  mapper: (field: Document | ObjectTypeNode) => void,
-) {
-  if (node.type === 'union') {
-    for (const scoped of node.of) {
-      mapObjectSplat(scoped, scope, mapper)
+  mapper: (attribute: ObjectAttribute) => ObjectAttribute,
+): ObjectTypeNode | UnknownTypeNode | NullTypeNode {
+  const attributes: Record<string, ObjectAttribute> = {}
+
+  const concreteNode = mapConcrete(node, scope, (node) => node)
+  if (concreteNode.type === 'object') {
```

Comment:
Hmm. Doesn't this mean that if this returns `false` we end up returning `{type: 'object', attributes: {}}` below? What about we instead wrap all of this in `mapConcrete`?

```ts
return mapConcrete(node, scope, (node) => {
  if (node.type !== 'object') return {type: 'unknown'}
  const attributes: Record<string, ObjectAttribute> = {}
  // same same…
})
```

Reply:
I did another pass, moved the handling of the object splats out in its own methods